src/main/java/com/deavensoft/paketomat/city/CityController.java

```diff
@@ -48,33 +50,54 @@ public class CityController {
     @GetMapping
     @Operation(summary = "Get cities", description = "Get all cities")
     @ApiResponse(responseCode = "200", description = "All cities are returned")
-    public List<City> getAllCities() {
-        return cityServiceImpl.getAllCities();
+    public List<CityDto> getAllCities() {
+       List<City> cities = cityServiceImpl.getAllCities();
+       List<CityDto> cityDtos = new ArrayList<>();
+
+       for (City city : cities) {
+           cityDtos.add(mapper.cityToCityDto(city));
+       }
+       log.info("All cities are returned");
+       return cityDtos;
     }
 
     @PostMapping
     @Operation(summary = "Add new city", description = "Add new city to the database")
     @ApiResponse(responseCode = "200", description = "New city added")
-    public void save(@RequestBody City c) {
-        cityServiceImpl.save(c);
+    public void save(@RequestBody CityDto c) {
+        City city = mapper.cityDtoToCity(c);
+        cityServiceImpl.save(city);
     }
 
     @GetMapping(path = "/{id}")
     @Operation(summary = "Get city", description = "Get city with specified id")
     @ApiResponse(responseCode = "200", description = "City with specified id returned")
-    public Optional<City> findCityById(@PathVariable(name = "id") long id) {
-        return cityServiceImpl.findCityById(id);
+    public CityDto findCityById(@PathVariable(name = "id") long id) throws NoSuchCityException {
+        Optional<City> city = cityServiceImpl.findCityById(id);
+
+        if (city.isEmpty()) {
+            throw new NoSuchCityException("There is no city with id " + id, HttpStatus.OK, 200);
+        }else {
+            City city1 = city.get();
+            CityDto cityDto = mapper.cityToCityDto(city1);
+            String mss = "City with id " + id + "is retuned";
+            log.info(mss);
+            return cityDto;
+        }
+
     }
 
     @DeleteMapping(path = "/{id}")
     @Operation(summary = "Delete city", description = "Delete city with specified id")
     @ApiResponse(responseCode = "200", description = "City with specified id deleted")
-    public int deleteCity(@PathVariable(name = "id") Long id) {
-        Optional<City> c = findCityById(id);
+    public int deleteCity(@PathVariable(name = "id") Long id) throws NoSuchCityException {
+        Optional<City> c = cityServiceImpl.findCityById(id);
+
         if (c.isEmpty())
-           log.info("There is no city with id:" + id);
+            throw new NoSuchCityException("There is no city with id " + id, HttpStatus.OK, 200);
         if(!c.isEmpty()) {
```

Comment:
Please, fix if statement. It can be fixed using if..else statement.

Reply:
Done

Reply:
Excellent :)